### PR TITLE
feature(tag) add a thread pool size for tags list

### PR DESCRIPTION
**Pull request by bitsf**
fix https://github.com/goharbor/harbor/issues/13098
add a thread pool size for tags list, in case it takes up too much resources

### CODE REVIEW

**Comment by reasonerjt** [comment]
Could you also check the setting on redis to understand the limit? it may be a bottleneck for pulling images concurrently.

File: src/core/api/repository.go [hunk 1]
```diff
@@ -649,6 +649,7 @@ func assembleTagsInParallel(client *registry.Repository, projectID int64, reposi
 		}
 	}
 
+	limit := make(chan bool, 100)
```
Comment (steven-zou) [inline src/core/api/repository.go:652]: 100 is determined by the DB connection max number?

**Review by steven-zou — APPROVED** [review]
LGTM

File: src/core/api/repository.go (hunk 1 above)
Comment (bitsf) [reply src/core/api/repository.go:652]: just an experience value

**Review by heww — APPROVED** [review]

File: src/core/api/repository.go [hunk 2]
```diff
@@ -659,6 +660,7 @@ func assembleTagsInParallel(client *registry.Repository, projectID int64, reposi
 			tag,
 			config.WithNotary(),
 			signatures,
+			limit,
```
Comment (ywk253100) [inline src/core/api/repository.go:663]: Just a minor suggestion, is it better to get the `true` from `limit` before starting a goroutine? In this way only 100 goroutines will be created

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**